security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -122,31 +107,35 @@ type SecretManagerClient struct {
 	// outputMutex protects writes of certificates to disk
 	outputMutex sync.Mutex
 
+	// Dynamically configured Trust Bundle Mutex
+	configTrustBundleMutex sync.RWMutex
+	// Dynamically configured Trust Bundle
+	configTrustBundle []byte
+
 	// queue maintains all certificate rotation events that need to be triggered when they are about to expire
 	queue queue.Delayed
 	stop  chan struct{}
 }
 
 type secretCache struct {
-	mu             sync.RWMutex
-	workload       *security.SecretItem
-	root           []byte
-	rootExpiration time.Time
+	mu       sync.RWMutex
+	workload *security.SecretItem
+	// currentCSRroot: Needed to compare old and new roots when workload is empty
+	currentCertroot []byte
 }
 
 // GetRoot returns cached root cert and cert expiration time. This method is thread safe.
-func (s *secretCache) GetRoot() (rootCert []byte, rootCertExpr time.Time) {
```

Comment:
So you are getting rid of the rootCertExpriration. I think this is fine. The ExpireTime is useful only for workload certs. But can you take care of all the places are working correctly, like this? https://github.com/istio/istio/blob/master/security/pkg/nodeagent/cache/secretcache.go#L584

Reply:
Not clear what additional changes are required  here (rotateTime). That function is mainly used to determine the rotation offset time for workload certs and is only invoked when we register a Secret - which also only happens only for workload certs.